Home_work15/Home_work15.cpp

```diff
@@ -0,0 +1,88 @@
+﻿#include <iostream>
+#include <string>
+#include <cmath>
+
+
+class Rectangle {
+public:
+	static int count() {
+		static int counter = 0;
+		return counter++;
+	};
+	float leght{}, hight{};
+	float getArea() {
+		return leght * hight;
+	}
+	float getPerimeter() {
+		return 2 * (leght + hight);
+	}
+	Rectangle(float leght, float hight) : leght{ leght }, hight{ hight } {
+		count();
+	}
+	Rectangle() {
+		leght = 10;
+		hight = 8;
+		count();
+	}
+};
+
+class Vector2D {
+public:
+	float x = 0.0;
+	float y = 0.0;
+	Vector2D(float x, float y) : x{ x }, y{ y } {}
+	Vector2D& operator=(const Vector2D& secondVector) {
+		this->x = secondVector.x;
+		this->y = secondVector.y;
+		return *this;
+	}
+	Vector2D operator+(const Vector2D& secondVector) {
+		this->x += secondVector.x;
+		this->y += secondVector.y;
+		return *this;
+	}
+	Vector2D operator-(const Vector2D& secondVector) {
+		this->x -= secondVector.x;
+		this->y -= secondVector.y;
+		return *this;
+	}
+
+	friend Vector2D operator+(const Vector2D& firstVector, const Vector2D& secondVector) {
+
+		return firstVector + secondVector;
+	}
+	friend Vector2D operator-(const Vector2D& firstVector, const Vector2D& secondVector) {
+
+		return firstVector - secondVector;
+	}
+	float operator()() {
+		return sqrt(x * x + y * y);
+	}
+	float& operator[](int i) {
+		if (i == 0) return x;
+		else if (i == 1) return y;
```

Comment:
not all test-cases are covered. The compiler should've warned you.
And what about i == 2?